### PR TITLE
Adds notes on how to handle subfolders in /pages

### PR DESCRIPTION
**Pull request by gregorskii**
Adds an example of how one can use subfolders in the /pages directory for root level routes, while still maintaining nested pages.

### CODE REVIEW

File: docs/docs/creating-and-modifying-pages.md [hunk 1]
```diff
@@ -183,3 +183,57 @@ exports.onCreatePage = async ({ page, boundActionCreators }) => {
...
+
+This code will ensure all root pages can have subfolders (which can contain other components, styles, etc)  in one logical place while still allowing for sub pages to be under them.
+
+Example folder structures:
+
+`/pages/index/index.js` -> becomes / due to extra if statement above
+`/pages/foo/index.js` -> becomes /foo/
+`/pages/foo/bar/index.js` -> becomes /foo/bar/
+`/pages/foo/bar/baz.js` -> becomes /foo/bar/baz/
+`/pages/foo/bar/zee/bar.js` -> becomes /foo/bar/zee/bar/zee
```
Comment (benjaminhoffman) [inline docs/docs/creating-and-modifying-pages.md]: wouldn't this be `/foo/bar/zee/bar/` ... where's the extra ending `zee` come from?

**Comment by benjaminhoffman** [comment]
@gregorskii this seems to work out of the box for me without this extra config.  maybe i am missing something?  for example, we have:
`pages/foo/index.js` --> `/foo`
`pages/bar/foo.js --> `/bar/foo`

**Comment by gregorskii** [comment]
@benjaminhoffman, yep I just tested it further again. It appears pages work correctly when they are nested in folders, however the index does not.

For instance `/pages/foo/index.js` will create a page `/foo`, however `pages/index/index.js` will not. The page will remain as /index.

To fix that I left this in place:

```
exports.onCreatePage = async ({ page, boundActionCreators }) => {
  const { createPage } = boundActionCreators;

  return new Promise((resolve, reject) => {
    if (/index/.test(page.path)) {
      page.path = '/';
    }

    createPage(page);
    resolve();
  });
};
```

Is there a shortcut for making the homepage work, but live in a folder?

Thanks,

Greg

File: docs/docs/creating-and-modifying-pages.md (hunk 1 above)
Comment (gregorskii) [reply docs/docs/creating-and-modifying-pages.md]: Think it was a typo.

**Comment by benjaminhoffman** [comment]
interesting.  I think the homepage (index route) is a unique case.  But for all others, `pages/foo/index.js` should suffice.

My comment is in regards to this: `You may wish to keep the root pages in a subfolder. You can do so by editing the `onCreatePage` function in `gatsby-node.js`.` . I don't think we need to update config when this functionality works out of the box.  I wonder what the use case is for doing `/pages/index/index.js`.  Any thoughts?

Also, keep in mind that any js/jsx file in the pages dir/subdir must resolve to a react component or string and it will be turned into a route (with regards to your comment about subfolders).

**Comment by gregorskii** [comment]
Agreed I think given the new info it’s not necessary to do this for any other route but the index.

The reason I typically do this is I like to keep module scss next to the component.

For index it would be:

/pages/index/index.jsx
/pages/index/index.module.scss

It is odd that the index is redundant, I typically would call this page “home” or “landing” and not index. I typically keep all of my compnents in a folder with an index.jsx to keep the scss with the component.

At this point I can close the PR unless you feel it would be helpful to document the index path renaming.

Thanks

**Comment by benjaminhoffman** [comment]
I'm indifferent but must defer to the core maintainers of gatsby.  Let's see what they say when this get to this PR.

I agree w your style of colocation.

**Comment by gregorskii** [comment]
Ok no worries, I’ll update the content of the pr to reflect the new notes when I am on a computer.

Thanks

**Comment by calcsam** [comment]
This seems like a valid use case, but rather than adding docs to this workaround, if we want to support a `/index/` folder we should add support for this in core. Thanks for the clarification @benjaminhoffman  and @gregorskii. Closing this.

**Comment by calcsam** [comment]
@gregorskii -- if you'd like to help out with a PR on core doing this, happy to help!

**Pull request closed without merging**